### ISSUE TITLE
Window needs a border if shadows are turned off [windows]

### ISSUE THREAD

**Issue by albinekb** [desktop/desktop#1585]
### Description
![image](https://cloud.githubusercontent.com/assets/5027156/26180162/17267ce6-3b68-11e7-8df9-1037fb56bbc6.png)

When shadows are turned off in windows, the window is hard to see.

### Version

**GitHub Desktop version:** 0.5.3

### Steps to Reproduce
1. Open the start menu
1. Search for "performance"
1. Open the app named "Adjust the appearance and performance of "Windows"
1. Choose "Adjust for best performance"

![image](https://cloud.githubusercontent.com/assets/5027156/26180178/353540fa-3b68-11e7-9c95-338a508c5e4a.png)

This will turn off shadows, and since the window is white, you cannot see where it ends.

Is there a way to check if shadows are turned off in electron? 🤔 Otherwise I'll open an issue there, since it could be useful to have such a feature.

**Expected behaviour:**
The window should have a border.

**Actual behaviour:**
The window is hard to see against white background.

**Comment by shiftkey** [comment]
Some initial thoughts:

 - I gather this is something we should fix in the app, because Electron Has No Feels On This (cc @desktop/electron to confirm)
 - we'd likely need to touch some OS API to test for whether this is has been set (the "Show shadows under windows" checkbox is the one we care about)
 - have we thought about a border-esque design given this accessibility issue @donokuda @niik?
 - is there a macOS equivalent where you can disable window shadows like this @joshaber?

**Comment by kevinsawicki** [comment]
> I gather this is something we should fix in the app, because Electron Has No Feels On This

Yeah, if there is an API to get this information or know when this OS setting has changed it would seem worth adding to https://electron.atom.io/docs/api/system-preferences/

**Comment by albinekb** [comment]
Maybe this is it: https://msdn.microsoft.com/en-us/library/windows/desktop/ms724947(v=vs.85).aspx#SPI_GETDROPSHADOW

🤔

**Comment by shiftkey** [comment]
@albinekb maybe? I'd have to remember how we did detection in WPF because I think we had similar work to do with the classic app.

**Comment by rlabrecque** [comment]
Even a 1 pixel border could/would probably be great design wise with or without shadow. Easier to do than modifying electron and having specific behavior.

**Comment by joshaber** [comment]
>is there a macOS equivalent where you can disable window shadows like this @joshaber?

LOL no that's crazy talk.

**Comment by shiftkey** [comment]
Paging @donoku and @niik for general feels before I return to the C++ mines

**Comment by niik** [comment]
![image](https://cloud.githubusercontent.com/assets/634063/26200743/f1249132-3bcf-11e7-9bb3-a82540c3ae51.png)

Okay, we're definitely renaming @donokuda to @donoku 

![image](https://cloud.githubusercontent.com/assets/634063/26201154/e181f8a8-3bd1-11e7-993d-e5cd9dcbe8d1.png)

Anyway, yes, we can totally add a border when normal shadows are disabled but I would like to have some more data on how common this is first since it appears to be a non-trivial effort to determine whether shadows are drawn or not. I image this isn't a terribly common thing but I have little data to back that up.

If someone would like to open a PR to add support for this in electron that'd be great.

**Comment by shiftkey** [comment]
> If someone would like to open a PR to add support for this in electron that'd be great.

If I can't get to this through the inbuilt system preferences I'll put together a PR.

**Comment by JonSaffron** [comment]
Anyone using a basic or classic desktop theme within windows will experience this problem. I agree with the comment above, it's well worth just adding a very thin border in all cases.

**Comment by aleksey-sh** [comment]
Does anyone know if this is being worked on? I am working on Windows Server 2012 R2 and its really pain to not to see the application border. You have no idea how many times a day I accidentally switch applications just because I wanted to move Github Desktop window. 

Unfortunately for some reason I have no shadows visible even if option is enabled in Visual Effects. If someone has some solution for that it would be greatly appreciated. Until then I vote for some kind of a visual clue to where the application border is.

Below screenshots are to demonstrate some of the issues

Anyone can point me to where the border is so that I can resize?
Anyone can say whether this is a Github Desktop on the google.com site?
![githubdesktop](https://user-images.githubusercontent.com/26824059/41874223-4aa1d460-7895-11e8-9e2f-3328e48d155b.png)

When I have multiple Github.com windows open the site's header blends in with the application header with no border.
![githubdesktop2](https://user-images.githubusercontent.com/26824059/41874869-1cf39f42-7897-11e8-9cbc-9ee6b12f5e47.png)

Thanks

**Comment by evecalm** [comment]
Any news?

**Comment by aleksey-sh** [comment]
I've just noticed that the README.md includes a screenshot that shows a border. However my latest installed version still missing a border. I have version 1.4.3 installed. Am I missing something? Is there an option or switch I have to change?

**Comment by JannesMeyer** [comment]
I'm wondering if there are any news on this issue?

Visual Studio Code (another Electron app) offers a setting to enable native title bars/window borders:

![image](https://user-images.githubusercontent.com/704336/66261694-83174700-e7d2-11e9-981d-8962f1c3fd3a.png)

**Comment by ahdung** [comment]
Need native window mode.

**Comment by jimetevenard** [comment]
Any updates from Github team on this?

This is still very annoying.

**Comment by BrendaHutchinson** [comment]
Hi i am the real Brenda Hutchinson,  and can not enter github hackers have
blocked me from my windows phone and account and here, can you fix this, or
tell me how? I appreciate your time and effort, thank you, sincerely Brenda
Hutchinson

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/15554.

**Comment by steveward** [comment]
Requested in https://github.com/desktop/desktop/issues/15896.

**Comment by yacoco00000** [comment]
> ### Description
> ![image](https://cloud.githubusercontent.com/assets/5027156/26180162/17267ce6-3b68-11e7-8df9-1037fb56bbc6.png)
> 
> When shadows are turned off in windows, the window is hard to see.
> 
> ### Version
> **GitHub Desktop version:** 0.5.3
> 
> ### Steps to Reproduce
> 1. Open the start menu
> 2. Search for "performance"
> 3. Open the app named "Adjust the appearance and performance of "Windows"
> 4. Choose "Adjust for best performance"
> 
> ![image](https://cloud.githubusercontent.com/assets/5027156/26180178/353540fa-3b68-11e7-9c95-338a508c5e4a.png)
> 
> This will turn off shadows, and since the window is white, you cannot see where it ends.
> 
> Is there a way to check if shadows are turned off in electron? 🤔 Otherwise I'll open an issue there, since it could be useful to have such a feature.
> 
> **Expected behaviour:** The window should have a border.
> 
> **Actual behaviour:** The window is hard to see against white background.

**Comment by RokeJulianLockhart** [comment]
Can someone ban @Achayou from the repository, please?
